backend/src/app/rpc/mutations/files.clj

```diff
@@ -110,16 +112,29 @@
 ;; --- Mutation: Set File shared
 
 (declare set-file-shared)
+(declare unlink-files)
+(declare absorb-library)
 
 (s/def ::set-file-shared
   (s/keys :req-un [::profile-id ::id ::is-shared]))
 
 (sv/defmethod ::set-file-shared
-  [{:keys [pool] :as cfg} {:keys [id profile-id] :as params}]
+  [{:keys [pool] :as cfg} {:keys [id profile-id is-shared] :as params}]
   (db/with-atomic [conn pool]
     (files/check-edition-permissions! conn profile-id id)
+    (when-not is-shared
+        (absorb-library conn params)
+        (unlink-files conn params))
     (set-file-shared conn params)))
 
+(def sql:unlink-files
+  "delete from file_library_rel
+    where library_file_id = ?")
+
+(defn- unlink-files
+  [conn {:keys [id] :as params}]
+  (db/exec-one! conn [sql:unlink-files id]))
```

Comment:
Can be done with simplier `db/delete!` helper instead of ad-hoc sql in this case.

Reply:
done